packages/ra-core/src/i18n/Translate.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { useTranslate } from './useTranslate';
+
+export const Translate = ({
+    i18nKey,
+    args,
+    children,
+    empty = 'no translation',
+    component = 'span',
+}: TranslateProps) => {
+    const translate = useTranslate();
+    const translatedMessage = translate(i18nKey, args);
+
+    if (translatedMessage && translatedMessage !== i18nKey) {
+        return React.createElement(component, {}, translatedMessage);
+    }
+    if (children) {
+        return React.createElement(component, {}, children);
+    }
+    if (empty) {
+        return React.createElement(component, {}, empty);
+    }
+    return null;
```

Comment:
I wouldn't even put it inside anything but a `Fragment` like we do for `ValidationError`:

```tsx
if (translatedMessage && translatedMessage !== i18nKey) {
        return <>{translatedMessage}</>;
}
```